appkt/src/main/java/com/richpathanimator/sample/CompoundViewSamplesActivity.kt

```diff
@@ -16,24 +17,26 @@ class CompoundViewSamplesActivity : AppCompatActivity() {
         val greenPath = colorPickerRichPathView.findRichPathByName("greenPath")
         val purplePath = colorPickerRichPathView.findRichPathByName("purplePath")
 
-        colorPickerRichPathView.setOnPathClickListener { clickedRichPath ->
-
-            bluePath?.strokeAlpha = 0f
-            redPath?.strokeAlpha = 0f
-            greenPath?.strokeAlpha = 0f
-            purplePath?.strokeAlpha = 0f
-
-            if (clickedRichPath == bluePath
-                    || clickedRichPath == redPath
-                    || clickedRichPath == greenPath
-                    || clickedRichPath == purplePath) {
-                clickedRichPath.strokeAlpha = 0.5f
-                showToast(clickedRichPath.name)
+        colorPickerRichPathView.onPathClickListener = object : RichPath.OnPathClickListener {
+            override fun onClick(richPath: RichPath) {
+                bluePath?.strokeAlpha = 0f
+                redPath?.strokeAlpha = 0f
+                greenPath?.strokeAlpha = 0f
+                purplePath?.strokeAlpha = 0f
+
+                if (richPath == bluePath
+                        || richPath == redPath
+                        || richPath == greenPath
+                        || richPath == purplePath) {
+                    richPath.strokeAlpha = 0.5f
+                    showToast(richPath.name)
+                }
             }
         }
     }
 
-    private fun showToast(msg: String) {
+    private fun showToast(msg: String?) {
+        msg ?: return
```

Comment:
I think this check should be done by the caller, it's easier for debugging to force the caller to not pass `null`, then the caller will be aware that his/her toast doesn't show because the `if condition` result is false
```kotlin
if(string != null) { showToast(string) }
//or any kotlin format
string?.let { showToast(string) }
```

Reply:
I agree your opinion, thanks.
Please see the fc3b44d.